File: Studnt-springboot/demo/src/main/java/com/example/demo/Entity/Student.java
package com.example.demo.Entity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table (name = "Student")

public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private int roll;
    private String email;

    @OneToOne(mappedBy = "student")
    private Subject subject;

}

File: student-angular/src/app/Student.ts
import {Subject} from "./Subject";

export class Student {
  id: number;
  name: string;
  roll: number;
  email: string;
  subjects: Subject[];

  constructor() {
    this.id = 0;
    this.name = '';
    this.roll = 0;
    this.email = '';
    this.subjects = [];
  }
}
